### PR TITLE
Add roxygen documentation for categorical plot function

### PR DESCRIPTION
**Pull request by nichowu**

### CODE REVIEW

File: R/categorical_eda.R [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                            color = NULL,
+                            title = NULL,
+                            font_size = 10,
+                            color_scheme = "Tableau 20",
+                            plot_height = 150,
+                            plot_width = 200,
+                            opacity = 0.6,
+                            facet = NULL) {
+    
+}
```
Comment (nichowu) [inline R/categorical_eda.R]: Add return: categorical_plot

File: R/categorical_eda.R [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#' options include "histogram" and "density".
+#' @param color An optional string variable used to set the color of the plot
+#' @param title An optional string variable used to set the title of the plot
+#' @param font_size An optional integer variable used to set the font size
+#' @param color_scheme An optional string variable used to set the color shceme
+#' @param plot_height An optional integer variable used to specify plot height
+#' @param plot_width An optional integer variable used to specify plot width
+#' @param opacity An optional integer variable used to specify density fill 
+#' opacity for the density plot
+#' @param facet An optional string variable used to specify facet factor
```
Comment (nichowu) [inline R/categorical_eda.R]: Add facet_col

File: R/categorical_eda.R [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#' Categorical variable plotting
+#' 
+#' This function takes in a data frame object and one categorical
+#' feature, to produce a histogram plot that visualizes the
+#' distribution of the feature. User can also choose to plot density
+#' graph of the feature by specifing in plot_type. 
+#' The function also offers customization on color, plot title, 
+#' font size, color-scheme, plot size and other common configurations.
+#'
+#' @param data A dataframe
```
Comment (nichowu) [inline R/categorical_eda.R]: A tibble or a dataframe

File: R/categorical_eda.R [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#' Categorical variable plotting
+#' 
+#' This function takes in a data frame object and one categorical
```
Comment (nichowu) [inline R/categorical_eda.R]: dataframe

File: R/categorical_eda.R [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#' feature, to produce a histogram plot that visualizes the
+#' distribution of the feature. User can also choose to plot density
+#' graph of the feature by specifing in plot_type. 
+#' The function also offers customization on color, plot title, 
+#' font size, color-scheme, plot size and other common configurations.
+#'
+#' @param data A dataframe 
+#' @param xval A string variable used to represent the x-axis.
```
Comment (nichowu) [inline R/categorical_eda.R]: A character variable ...

File: R/categorical_eda.R [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+#' @param yval An optional string variable used to represent the y-axis
```
Comment (nichowu) [inline R/categorical_eda.R]: get rid of yval?

File: R/categorical_eda.R [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+#' @param plot_type An optional string variable used to specify plot type. 
+#' options include "histogram" and "density".
+#' @param color An optional string variable used to set the color of the plot
```
Comment (nichowu) [inline R/categorical_eda.R]: character variable...

set the color variable of the plot

File: R/categorical_eda.R (hunk 2 above)
Comment (nichowu) [inline R/categorical_eda.R]: character variable

File: R/categorical_eda.R [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+#' @param title An optional string variable used to set the title of the plot
+#' @param font_size An optional integer variable used to set the font size
+#' @param color_scheme An optional string variable used to set the color shceme
+#' @param plot_height An optional integer variable used to specify plot height
+#' @param plot_width An optional integer variable used to specify plot width
+#' @param opacity An optional integer variable used to specify density fill 
+#' opacity for the density plot
+#' @param facet An optional string variable used to specify facet factor
+#'
+#' @return A ggplot2 object
```
Comment (nichowu) [inline R/categorical_eda.R]: @return A ggplot2 object, either a histogram or a density plot

File: R/categorical_eda.R [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+#' categorical_plot <- categorical_eda(data = df, 
+#'                                     xval = "Origin", 
+#'                                     color = "Horsepower", 
+#'                                     title = "Histogram of Origin in Different Levells of Horsepower",
+#'                                     plot_height = 100,
+#'                                     plot_weight = 200,
+#'                                     facet = "Country")
+#' 
+#' 
+categorical_eda <- function(data,
```
Comment (nichowu) [inline R/categorical_eda.R:49]: get rid of yval

File: R/categorical_eda.R [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+#' @examples 
+#' 
+#' df <- data(mtcars)
+#' categorical_plot <- categorical_eda(data = df, 
+#'                                     xval = "Origin", 
+#'                                     color = "Horsepower", 
+#'                                     title = "Histogram of Origin in Different Levells of Horsepower",
+#'                                     plot_height = 100,
+#'                                     plot_weight = 200,
+#'                                     facet = "Country")
```
Comment (nichowu) [inline R/categorical_eda.R]: facet = "Origin"

File: R/categorical_eda.R [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+#' 
+#' 
+categorical_eda <- function(data, 
+                            xval = NULL,
```
Comment (nichowu) [inline R/categorical_eda.R]: for required argument, get rid of NULL.

**Review by nichowu — COMMENTED** [review]
Change requested by team members.

File: R/categorical_eda.R [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+#' feature, to produce a histogram plot that visualizes the
+#' distribution of the feature. User can also choose to plot density
+#' graph of the feature by specifying in plot_type. 
+#' The function also offers customization on color, plot title, 
+#' font size, color-scheme, plot size and other common configurations.
+#'
+#' @param data A tibble or a dataframe 
+#' @param xval A character variable used to represent the x-axis.
+#' @param plot_type An optional character variable used to specify plot type. 
+#' options include "histogram" and "density".
```
Comment (ChuckHo777) [inline R/categorical_eda.R]:
```suggestion
#' Options include "histogram" and "density".
```

File: R/categorical_eda.R [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+#'                                     title = "Histogram of Origin in Different Levells of Horsepower",
+#'                                     plot_height = 100,
+#'                                     plot_weight = 200,
+#'                                     facet_factor = "Origin",
+#'                                     facet_col = 2)
+#' 
+#' 
+categorical_eda <- function(data, 
+                            xval,
+                            plot_type "histogram",
```
Comment (ChuckHo777) [inline R/categorical_eda.R]:
```suggestion
                            plot_type = "histogram",
```

**Review by ChuckHo777 — APPROVED** [review]
Good!

**Review by ChuckHo777 — APPROVED** [review]

**Pull request merged**